frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -91,7 +91,14 @@ export const createOrUpdateImageStream = (
     },
   };
 
-  const imageStream = _.merge({}, originalImageStream || {}, newImageStream);
+  const imageStream = {
+    ...originalImageStream,
+    ...newImageStream,
+    metadata: {
+      ...(originalImageStream ? originalImageStream.metadata : newImageStream.metadata),
+      labels: newImageStream.metadata.labels,
+    },
+  };
```

Comment:
as this logic seems repeated, will it make sense to have util which can be reused?
```
getMergedData(originalData, newData): modifiedData {
 return {
    ...originalData,
    ...newData,
    metadata: {
      ...(originalData ? originalData.metadata : newData.metadata),
      labels: newData.metadata.labels,
    },
  }
}
....
const imageStream =  getMergedData(originalData, newData): modifiedData
```

Reply:
I have created the util `mergeData` and have placed it in the  `resource-label-utils.ts` file as suggested by you. I had tried placing the util in the `shared-submit-utils.ts` but then when I import it in the `create-knative-utils.ts` file, the page fails to load with a TypeError `Object(...) is not a function`

Reply:
yes, am not sure why we are getting the error if add utils in `shared-submit-utils.ts` and not in  `resource-label-utils.ts` @christianvogt  maybe able to help /add